OasisDEX/oasis-borrow
# Reusable dpm in flow state component

# [Reusable dpm in flow state component](https://app.shortcut.com/oazo-apps/story/11298/update-empty-dpm-check-in-flow-component)

<please insert a shortcut link above>
  
## Changes 👷‍♀️

- Updated `getFlowStateConfig` hook with optional param that allows for custom determination if proxy is consumed or not,
- created `getFlowStateFilter` function for filtering DPMs created by user that could be reused for different positions.
  
## How to test 🧪

Turn on `AjnaReusableDPM` feature flag.

Creating new Ajna position, creation of new DPM should not be required whenever:
- there is an existing AAVE position on it,
- there is an existing Ajna position on it, but on different pool,
- there is an existing Ajna position on it, on same pool, but of different type (for this purpose Borrow and Mutliply are considered as the same borrowish type).
